Sofa/framework/Helper/src/sofa/helper/Utils.cpp

```diff
@@ -226,6 +233,111 @@ std::map<std::string, std::string> Utils::readBasicIniFile(const std::string& pa
     return map;
 }
 
+// no standard/portable way
+const std::string& Utils::getUserLocalDirectory()
+{
+
+    auto computeUserHomeDirectory = []()
+    {
+// Windows: "LocalAppData" directory i.e ${HOME}\AppData\Local
+#ifdef WIN32
+        std::wstring wresult;
+        wchar_t* path = nullptr;
+        const auto hr = SHGetKnownFolderPath(FOLDERID_LocalAppData, 0, nullptr, &path);
+        if (SUCCEEDED(hr))
+        {
+            wresult = std::wstring(path);
+        }
+        if (path)
+        {
+            CoTaskMemFree(path);
+        }
+
+        // convert wide string (unicode) to standard string
+        std::string result(wresult.length(), 0);
+        std::transform(wresult.begin(), wresult.end(), result.begin(), [](wchar_t c) {
+            return (char)c;
+        });
+
+        return result;
+
+#elif defined(__APPLE__) // macOS : ${HOME}/Library/Application Support
+        // https://stackoverflow.com/questions/5123361/finding-library-application-support-from-c
+        
+        char path[PATH_MAX];
+        auto state = sysdir_start_search_path_enumeration(SYSDIR_DIRECTORY_APPLICATION_SUPPORT,
+                                                          SYSDIR_DOMAIN_MASK_USER);
+        if ((state = sysdir_get_next_search_path_enumeration(state, path)))
+        {
+            glob_t globbuf;
+            if (glob(path, GLOB_TILDE, nullptr, &globbuf) == 0) 
+            {
+                std::string result(globbuf.gl_pathv[0]);
+                globfree(&globbuf);
+                return result;
+            } 
+            else
+            {
+                // "Unable to expand tilde"
+                return std::string("");
+            }
+        }
+        else
+        {
+            // "Failed to get settings folder"
+            return std::string("");
+        }
+        
+#else // Linux: either ${XDG_CONFIG_HOME} if defined, or ${HOME}/.config (should be equivalent)
+        const char* configDir;
+
+        // if env.var XDG_CONFIG_HOME is defined
+        if ((configDir = std::getenv("XDG_CONFIG_HOME")) == nullptr)
+        {
+            const char* homeDir;
+
+            // else if HOME is defined
+            if ((homeDir = std::getenv("HOME")) == nullptr)
+            {
+                // else system calls are used
+                homeDir = getpwuid(getuid())->pw_dir;
+            }
+
+            return std::string(homeDir) + std::string("/.config");
+        }
+        else
+        {
+            return std::string(configDir);
+        }
+
+#endif
+    };
+
+    static std::string homeDir = FileSystem::cleanPath(computeUserHomeDirectory());
+    return homeDir;
+}
+
+const std::string& Utils::getSofaUserLocalDirectory()
+{
+    auto computeSofaLocalDirectory = []()
+    {
+        const std::filesystem::path localDirPath = getUserLocalDirectory() + "/SOFA";
```

Comment:
You can use `FileSystem::append` here, or `std::filesystem::append`, or `std::filesystem::path{getUserLocalDirectory()} / "SOFA"`